buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java

```diff
@@ -29,89 +32,151 @@
+    String function = declaration + " {} {}";
+    assertFunctionIsParsed(declaration, function);
+  }
+
+  @Test public void processFunctionEmptyLinesBetweenCurlyBlocks() {
+    String declaration = "void foo() :\n";
+    String function = declaration
+      + "\n"
+      + "{\n"
+      + "}\n"
+      + "\n"
       + "{\n"
-      + "    // overridden by intermediate\n"
       + "}";
-    String qux = "void qux(int arg1, int arg2) :\n"
+    assertFunctionIsParsed(declaration, function);
+  }
+
+  @Test public void processFunctionStringDeclaration() {
+    String declaration = "void foo() :\n";
+    String function = declaration
       + "{\n"
-      + "    String x = \" } \"\n"
+      + "    String x = \" } \";\n"
       + "}\n"
       + "{\n"
-      + "    // Below is a }\n"
       + "}";
-    String quux = "void quux(int arg1,\n"
-      + "    int arg2\n"
-      + ") :\n"
+    assertFunctionIsParsed(declaration, function);
+  }
+
+  @Test public void processFunctionCharacterDeclaration() {
+    String declaration = "void foo() :\n";
+    String function = declaration
       + "{\n"
-      + "    char x = ' } '\n"
-      + "    String y;\n"
+      + "    Character x = ' } ';\n"
       + "}\n"
       + "{\n"
-      + "    /* All invalid curly braces:\n"
-      + "    }\n"
+      + "}";
+    assertFunctionIsParsed(declaration, function);
+  }
+
+  @Test public void processFunctionContainsComments() {
+    String declaration = "void foo() :\n";
+    String function = declaration
+      + "{\n"
       + "    // }\n"
-      + "    \" } \"\n"
-      + "    ' } '\n"
-      + "    */\n"
-      + "    <TOKEN> {\n"
-      + "        y = \" { } } \"\n"
-      + "    }\n\n"
-      + "    // Not a string: \"\n"
+      + "}\n"
+      + "{\n"
+      + "    /* } */\n"
       + "}";
-    return new LinkedHashMap<String, String>() {{
-      put("foo", foo);
-      put("bar", bar);
-      put("baz", baz);
-      put("qux", qux);
-      put("quux", quux);
-    }};
+    assertFunctionIsParsed(declaration, function);
   }
 
-  @Test public void testExtractFunctions() {
-    DialectGenerate dialectGenerate = setupDialectGenerate();
-    Map<String, String> result = dialectGenerate.extractFunctions();
-    Map<String, String> expected = getExpectedExtractionResults();
-    assertEquals(expected.size(), result.size(),
-        "Resultant map size doesn't match expected map size");
-    for (String key : expected.keySet()) {
-      assertTrue(result.containsKey(key),
-          "Resultant map doesn't contain expected key: " + key);
-      assertEquals(expected.get(key), result.get(key));
-    }
+  @Test public void processFunctionMissingCurlyBlockFails() {
+    String declaration = "void foo() :\n";
+    String function = declaration
+      + "{\n"
+      + "}";
+    assertThrows(IllegalStateException.class,
+        () -> assertFunctionIsParsed(declaration, function));
+  }
+
+  @Test public void processFileEmpty() {
+    String testName = "empty";
```

Comment:
nit: Here and elsewhere, you can just omit the temporary variable and pass the name of the file in directly to assertFileProcessed().